stringutil/stringutil_test.go

```diff
@@ -454,158 +454,221 @@ func TestNumericOnly(t *testing.T) {
 	}
 }
 
-func TestObfuscateString(t *testing.T) {
+func TestObfuscate(t *testing.T) {
 	assert := assert1.New(t)
-	var testData = []struct {
-		str      string
-		expected string
-	}{
-		{
-			str:      "",
-			expected: "",
-		},
-		{
-			str:      "1",
-			expected: "*",
-		},
-		{
-			str:      "12",
-			expected: "1*",
-		},
-		{
-			str:      "123",
-			expected: "12*",
-		},
-		{
-			str:      "1234",
-			expected: "12**",
-		},
-		{
-			str:      "12345",
-			expected: "123**",
-		},
-		{
-			str:      "123456",
-			expected: "123***",
-		},
-		{
-			str:      "1234567",
-			expected: "123****",
-		},
-		{
-			str:      "spaces and characters !@#$%^&*()",
-			expected: "spa*****************************",
-		},
+	type testCase struct {
+		Name string
+
+		str        string
+		length     int
+		direction  string
+		obfuscator string
+
+		Expected string
 	}
-	for _, data := range testData {
-		assert.Equal(data.expected, ObfuscateString(data.str))
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			actual := Obfuscate(tc.str, tc.length, tc.direction, tc.obfuscator)
+
+			assert.Equal(tc.Expected, actual, tc.Name)
+		})
 	}
+
+	validate(t, &testCase{
+		Name: "Empty",
+
+		str:        "",
+		length:     0,
+		direction:  "left",
+		obfuscator: "*",
+
+		Expected: "",
+	})
+	validate(t, &testCase{
+		Name: "Obfuscate none",
+
+		str:        "foo bar",
+		length:     0,
+		direction:  "left",
+		obfuscator: "*",
+
+		Expected: "foo bar",
+	})
+	validate(t, &testCase{
+		Name: "Invalid direction",
+
+		str:        "foo bar",
+		length:     5,
+		direction:  "middle",
+		obfuscator: "*",
+
+		Expected: "foo bar",
+	})
+	validate(t, &testCase{
+		Name: "Empty obfuscator",
+
+		str:        "foo bar",
+		length:     5,
+		direction:  "left",
+		obfuscator: "",
+
+		Expected: "ar",
+	})
+	validate(t, &testCase{
+		Name: "Long obfuscator",
+
+		str:        "foober",
+		length:     3,
+		direction:  "left",
+		obfuscator: "bar",
+
+		Expected: "barbarbarber",
+	})
+	validate(t, &testCase{
+		Name: "Obfuscation longer than input string",
+
+		str:        "foo",
+		length:     5,
+		direction:  "left",
+		obfuscator: "X",
+
+		Expected: "XXXXX",
+	})
+	validate(t, &testCase{
+		Name: "Obfuscate left",
+
+		str:        "foo bar foo",
+		length:     6,
+		direction:  "left",
+		obfuscator: "*",
+
+		Expected: "******r foo",
+	})
+	validate(t, &testCase{
+		Name: "Obfuscate right",
+
+		str:        "this is redacted",
+		length:     8,
+		direction:  "right",
+		obfuscator: "█",
+
+		Expected: "this is ████████",
+	})
```

Comment:
it's funny, this seems like such a small change to how testing is done, but seems a lot cleaner to me. I dunno, I like it. :+1: